components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -459,10 +461,37 @@ export default class PluginManagement extends AdminSettings {
 
         this.setState({uploading: true});
 
-        const {error} = await this.props.actions.uploadPlugin(file);
-        this.setState({fileSelected: false, fileName: null, uploading: false, serverError: null});
+        const {error} = await this.props.actions.uploadPlugin(file, false);
+        this.setState({uploading: false, serverError: null});
         Utils.clearFileInput(element);
 
+        if (error) {
+            if (error.server_error_id === 'app.plugin.install_id.app_error') {
+                this.setState({showOverwritePluginModal: true});
+                return;
+            }
+            this.setState({file: null, fileSelected: false});
+
+            if (error.server_error_id === 'app.plugin.activate.app_error') {
+                this.setState({serverError: Utils.localizeMessage('admin.plugin.error.activate', 'Unable to upload the plugin. It may conflict with another plugin on your server.')});
+            } else if (error.server_error_id === 'app.plugin.extract.app_error') {
+                this.setState({serverError: Utils.localizeMessage('admin.plugin.error.extract', 'Encountered an error when extracting the plugin. Review your plugin file content and try again.')});
+            } else {
+                this.setState({serverError: error.message});
+            }
+        }
+    }
+
+    handleOverwritePluginCancel = async () => {
```

Comment:
I don't think `async` is required here, since we're not `await`ing on anything nor are we returning a promise.

Reply:
Done